samza-core/src/main/java/org/apache/samza/clustermanager/SamzaApplicationState.java

```diff
@@ -141,6 +143,36 @@
    */
   public final AtomicInteger redundantNotifications = new AtomicInteger(0);
 
+  /**
+   * Number of container allocations that proceeded because they met standby container constraints.
+   */
+  public final AtomicInteger successfulStandbyAllocations = new AtomicInteger(0);
+
+  /**
+   * Number of container allocations that were dis-allowed because they
+   * did not meet standby container constraints.
+   */
+  public final AtomicInteger failedStandbyAllocations = new AtomicInteger(0);
+
+  /**
+   * Number of active container failovers initiated in which a standby container was found.
+   * If two standby containers had to selected for one failing active, it counts as two.
+   */
+  public final AtomicInteger failoversToStandby = new AtomicInteger(0);
+
+  /**
+   * Number of active container failovers initiated in which a standby container was NOT found.
+  */
+  public final AtomicInteger failoversToAnyHost = new AtomicInteger(0);
+
+  /**
+   * Number of stops of standby containers that completed.
+   */
+  public final AtomicInteger standbyStopsComplete = new AtomicInteger(0);
+
+  // Map of active containers that are in failover, indexed by the active container's resourceID (at the time of failure)
+  public final ConcurrentMap<String, ContainerFailoverState> failovers = new ConcurrentHashMap<String, ContainerFailoverState>(0);
```

Comment:
move state that pertains to failover to a new class.

Reply:
Moved to ContainerFailoverState class